### PR TITLE
Do not pass a `random` sort filter from AutoSearchInput

### LINKED ISSUE #8298: Error message displayed when searching addons from recommended addons page
Steps to reproduce:
1.Go to addons.mozilla.org
2.Press the "See more recommended extensions" link from homepage
3. Search for an addon and press Enter

Expected results:
A list of search results is displayed

Actual results:
"The "sort" parameter "random" can only be specified when the "featured" or "recommended" parameter is also present, and the "q" parameter absent." error is displayed and 0 results are found.

Please see screenshot of the issue:
Screen Shot 2019-07-11 at 19 53 30

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #8298

### CODE REVIEW

**Comment by willdurand** [comment]
I think it'd make sense to let @kumar303 review this patch too (as I suggested a part of it).

File: src/amo/components/AutoSearchInput/index.js [hunk 1]
```diff
@@ -151,6 +152,13 @@ export class AutoSearchInputBase extends React.Component<InternalProps, State> {
       delete filtersFromLocation.page;
     }
 
+    if (
+      filtersFromLocation.sort &&
+      filtersFromLocation.sort === SEARCH_SORT_RANDOM
+    ) {
+      delete filtersFromLocation.sort;
```
Comment (kumar303) [inline src/amo/components/AutoSearchInput/index.js]: It's not obvious why this is getting deleted so I suggest putting a comment here.

**Comment by bobsilverberg** [comment]
> r+wc
> 
> It seems a little fragile to add a special case just for `AutoSearchInput` but I can't think of anything better. Are there any other places where `sort=random` might leak into a query and cause the same error?

I was a bit concerned myself at first, but there really isn't anywhere else it might be used, and `AutoSearchInput` is in fact the only place where a user-submitted query takes place, so I think it's fine.

**Comment by willdurand** [comment]
> I was a bit concerned myself at first, but there really isn't anywhere else it might be used, and `AutoSearchInput` is in fact the only place where a user-submitted query takes place, so I think it's fine.

It turns out that there is: https://github.com/mozilla/addons-frontend/issues/8301.

**Comment by bobsilverberg** [comment]
> > I was a bit concerned myself at first, but there really isn't anywhere else it might be used, and `AutoSearchInput` is in fact the only place where a user-submitted query takes place, so I think it's fine.
> 
> It turns out that there is: #8301.

Ah, because there are really two different rules in play that result in this error, and one has nothing to do with `q`.

**Pull request merged**